### ISSUE TITLE
Replace deprecated grpc-web server/client library

### ISSUE THREAD

**Issue by migmartri** [chainloop-dev/chainloop#172]
For the UI frontend we have enabled `grpc-web` in the backend via [improbable-eng/grpc-web](https://github.com/improbable-eng/grpc-web), which has been deprecated in favor of the official grpc-web support 

This means to a) find a way to do the translation in the server, b) potentially replace our current typescript bindings (generated using [improbable-eng/ts-protoc-gen](https://github.com/improbable-eng/ts-protoc-gen) and c) update the frontent code  

The good news is that the Kubeapps team have gone through the same process and have added great context on alternatives (thanks @absoludity) https://github.com/vmware-tanzu/kubeapps/issues/6013

Long story short would be to use https://github.com/bufbuild/connect-es as replacement.

 
```[tasklist]
### Tasks
- [ ] Controlplane API
- [ ] CAS API
- [ ] CLI client update
- [ ] Frontent Client update
```

**Comment by migmartri** [comment]
Also

```
cd ./api && buf generate
WARN    Plugin "buf.build/stephenh/ts-proto" is deprecated
```
